src/components/body/ProductQuantity.js

```diff
@@ -0,0 +1,126 @@
+import {useDispatch, useSelector} from "react-redux";
+import {set_product_counter, set_products, set_total_cost, set_total_items} from "../../rootReducer";
+import calculateTotal from "../../helpers";
+import './ProductQuantity.css';
+
+
+function CartClearButton({product}) {
+    const dispatch = useDispatch()
+    let products = useSelector(state => state.products)
+    let product_counter = useSelector(state => state.product_counter)
+    let total_items = useSelector(state => state.total_items)
+
+    async function clearFromCart(product) {
+        products[product.id].pop()
+        product_counter[product.id] -= 1
+        delete products[product.id]
+        delete product_counter[product.id]
+
+        dispatch({type: set_products, products: products})
+        dispatch({type: set_product_counter, product_counter: product_counter})
+        dispatch({type: set_total_items, total_items: --total_items})
+        dispatch({type: set_total_cost, total_cost: await calculateTotal(products)})
+    }
+
+    return (
+        //<span className="cart-product-quantity-delete" onClick={() => clearFromCart(products[id1][0])}>Delete</span>
+        <span id="clear-cart" onClick={() => clearFromCart(product)}>Delete</span>
+    )
+}
+
+
+function ProductQuantityDecrementer({product, quantity}) {
+    const dispatch = useDispatch()
+    let products = useSelector(state => state.products)
+    let total_items = useSelector(state => state.total_items)
+    let product_counter = useSelector(state => state.product_counter)
+
+    async function removeFromCart(product) {
+        dispatch({type: set_total_items, total_items: --total_items})
+        decrementCart(product)
+        dispatch({type: set_total_cost, total_cost: await calculateTotal(products)})
+      }
+
+    async function decrementCart(product) {
+        if(!(product.id in products)) {
+          product_counter[product.id] = 0
+        } else {
+            products[product.id].pop()
+            product_counter[product.id] -= 1
+        }
+        dispatch({type: set_products, products: products})
+        dispatch({type: set_product_counter, product_counter: product_counter})
+    }
+
+    return (
+        <div className="container-inc-dec" id="container-decrement">
+                  {
+                    (quantity < 2) ?
+                      <span id="decrement"> - </span>
```

Comment:
Hmm, yeah I guess I started converting some classNames to ids in cases where the id was actually unique. But then I realized that maybe that needs more consideration since most React code out there uses className as a standard regardless of uniqueness. I want to research it a bit more before I convert everything.